specs/pvs/corla.pvs

```diff
@@ -1650,6 +1772,186 @@ END system_assumptions
 
 %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
 
+%%% \section{System Logging}
+
+%%% The system must log a variety of events. One reason to log
+%%% information is to help understand how the system is operating, fix
+%%% bugs post-facto, understand how users are using the system,
+%%% etc. Another, more important reason to log in this context of this
+%%% system is to provide an indeliable record of administrator and
+%%% auditor actions so that any audit can be "replayed" by any third
+%%% party.
+
+logging: THEORY
+BEGIN
+  log: TYPE = sequence[string]
+  chain_of_custody_log: TYPE FROM log
+% @trace Per email from Jerome Lovato at CDOS on 19 July 2017: Logging
+% of Audit Board and SOS user errors during use of the system [...is
+% the only logging we care about].
+  rla_tool_log: TYPE FROM log
+END logging
+
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+
+%%% \section{Data Model}
```

Comment:
A picture of a more recent data model (which I don't think I've actually put online....), with HandContestBatch and HandVoteCount classes is visible in the doc/model_graph.png file, e.g. here: ![model_graph](https://user-images.githubusercontent.com/119472/28495296-5e2f86f2-6f05-11e7-9029-224275ad9fe7.png)